tests/test_controller.py

```diff
@@ -0,0 +1,428 @@
+import asyncio
+import time
+from typing import Optional
+
+import pytest
+from pydantic import BaseModel
+
+from browser_use.agent.views import ActionModel, ActionResult
+from browser_use.browser.browser import Browser, BrowserConfig
+from browser_use.browser.context import BrowserContext
+from browser_use.controller.service import Controller
+from browser_use.controller.views import (
+	ClickElementAction,
+	CloseTabAction,
+	GoToUrlAction,
+	InputTextAction,
+	NoParamsAction,
+	OpenTabAction,
+	ScrollAction,
+	SearchGoogleAction,
+	SwitchTabAction,
+)
+
+
+class TestControllerIntegration:
+	"""Integration tests for Controller using actual browser instances."""
+
+	@pytest.fixture(scope='module')
+	def event_loop(self):
+		"""Create and provide an event loop for async tests."""
+		loop = asyncio.get_event_loop_policy().new_event_loop()
+		yield loop
+		loop.close()
+
+	@pytest.fixture(scope='module')
+	async def browser(self, event_loop):
+		"""Create and provide a Browser instance with security disabled."""
+		browser_instance = Browser(
+			config=BrowserConfig(
+				headless=True,
+				disable_security=True,  # This disables web security features
+				browser_args=[
+					'--disable-web-security',
+					'--disable-features=IsolateOrigins,site-per-process',
+					'--disable-site-isolation-trials',
+					'--disable-features=BlockInsecurePrivateNetworkRequests',
+					'--disable-blink-features=AutomationControlled',
+				],
```

Comment:
```suggestion
```

we can remove all of these, patchright handles it all

we are going to remove the custom args from the library as well very soon